src/benchmarks/Sofa.Type/Quat.cpp

```diff
@@ -0,0 +1,180 @@
+#include <benchmark/benchmark.h>
+
+#include <utils/RandomValuePool.h>
+
+#include <sofa/type/Vec.h>
+#include <sofa/type/Quat.h>
+
+#include <array>
+#include <numeric>
+
+constexpr int64_t minSubIterations = 8 << 10;
+constexpr int64_t maxSubIterations = 8 << 10;
+
+static void BM_Quat_rotateVec(benchmark::State& state);
+BENCHMARK(BM_Quat_rotateVec)->RangeMultiplier(2)->Ranges({ {minSubIterations, maxSubIterations} })->Unit(benchmark::kMicrosecond);
+static void BM_Quat_rotateVec_impl_new(benchmark::State& state);
+BENCHMARK(BM_Quat_rotateVec_impl_new)->RangeMultiplier(2)->Ranges({ {minSubIterations, maxSubIterations} })->Unit(benchmark::kMicrosecond);
+static void BM_Quat_rotateVec_impl_old(benchmark::State& state);
+BENCHMARK(BM_Quat_rotateVec_impl_old)->RangeMultiplier(2)->Ranges({ {minSubIterations, maxSubIterations} })->Unit(benchmark::kMicrosecond);
+
+static void BM_Quat_invrotateVec(benchmark::State& state);
+BENCHMARK(BM_Quat_invrotateVec)->RangeMultiplier(2)->Ranges({ {minSubIterations, maxSubIterations} })->Unit(benchmark::kMicrosecond);
+
+void BM_Quat_rotateVec(benchmark::State& state)
+{
+    using Quat = sofa::type::Quatf;
+    using Vec3 = sofa::type::Vec3f;
+
+    constexpr auto totalsize = maxSubIterations * 3 + maxSubIterations * 4;
+    const auto& quatValues = RandomValuePool<float, totalsize>::get(0.f, 1.f);
+    const auto& vectValues = RandomValuePool<float, totalsize>::get();
+    std::vector<Quat> quat;
+    std::vector<Vec3> vect;
+
+    for (auto _ : state)
+    {
+        state.PauseTiming();
+        quat.resize(state.range(0));
```

Comment:
`state.range(0)` is constant, so I would have moved the initializations of the vectors out of the loop

Reply:
Really ? I thought it was the number of iterations for the given state
so it will change for each iteration of the for-loop

Reply:
`state.range(0)` is a parameter you give to your benchmark. Depending on the setup of your benchmark, multiple benchmarks are run with a different parameter.

Reply:
Note that pausing and resuming the timing has a high overhead, so it is not recommended to use them. Here you can initialize the vectors out of the loop

Reply:
> `state.range(0)` is a parameter you give to your benchmark. Depending on the setup of your benchmark, multiple benchmarks are run with a different parameter.

I got confused that state was given by the for-loop.... 🙈